processor/src/stack/mod.rs

```diff
@@ -89,9 +89,17 @@ impl Stack {
     /// the stack depth is less than 16, the un-used slots contain ZEROs.
     #[allow(dead_code)]
     pub fn trace_state(&self) -> [Felt; STACK_TOP_SIZE] {
+        self.trace_state_at_step(self.step)
+    }
+
+    /// Returns trace state at a specific step
+    ///
+    /// Trace state is always 16 elements long and contains the top 16 values of the stack. When
+    /// the stack depth is less than 16, the un-used slots contain ZEROs.
+    pub fn trace_state_at_step(&self, step: usize) -> [Felt; STACK_TOP_SIZE] {
```

Comment:
I'd probably rename this to something like `get_stack_state_at()` and would return a vector from here which contains the entire state of the stack (stack top + overflow).

Reply:
Do you think it makes sense to split it into stack and overflow?

Reply:
I think for these purposes we should merge them. From the "outside" standpoint, there should just a single view of the stack. The fact that it is split into two parts is an implementation detail.